server/graphql/v2/mutation/UploadedFileMutations.ts

```diff
@@ -59,34 +62,18 @@ const uploadedFileMutations = {
       }
 
       // Upload & parse files
-      const useOCR = args.files.some(r => r.parseDocument);
+      const canUseOCR = config.env.OC_ENV !== 'production' || req.remoteUser.isRoot();
+      const useOCR = canUseOCR && args.files.some(r => r.parseDocument);
       const parser = useOCR ? getExpenseOCRParser() : null;
       return Promise.all(
         args.files.map(async ({ file, kind, parseDocument }) => {
-          const uploadedFile = await models.UploadedFile.uploadGraphQl(await file, kind, req.remoteUser);
-          let parsingResult;
+          const result: UploadFileResult = { file: null, parsingResult: null };
+          result.file = await models.UploadedFile.uploadGraphQl(await file, kind, req.remoteUser);
           if (parseDocument) {
-            if (parser) {
-              try {
-                const [result] = await parser.processUrl(uploadedFile.url);
-                parsingResult = {
-                  success: true,
-                  confidence: result.confidence,
-                  expense: {
-                    description: result.description,
-                    amount: result.amount,
-                    incurredAt: result.date,
-                  },
-                };
-              } catch (e) {
-                parsingResult = { success: false, message: `Could not parse document: ${e.message}` };
-              }
-            } else {
-              parsingResult = { success: false, message: 'OCR parsing is not available' };
-            }
+            result.parsingResult = await runOCRForExpenseFile(parser, result.file);
```

Comment:
Cool, we moved the missing parser logic into the library and standardized the response. 👍🏼